Jenkinsfile

```diff
@@ -337,28 +337,41 @@ def setEnvForDBModule(openShiftHost, openShiftToken, svcName, projName, mysqlUse
 }
 def promoteServiceSetup(openShiftHost, openShiftToken, svcName,registry,imageNameSpace, tagName, projName) {
     try {
-    sh """ 
-        oc create dc ${svcName} --image=${registry}/${imageNameSpace}/${svcName}:${tagName} -n ${projName} 2> /dev/null  
-        oc deploy ${svcName} --cancel -n ${projName} 2> /dev/null 
-        oc expose dc ${svcName} --port=8080 -n ${projName} 2> /dev/null 
-        oc expose svc ${svcName} --name=${svcName} -n ${projName} 2> /dev/null 
-    """
+        sh """
+            oc delete dc ${svcName} -n ${projName} 2> /dev/null
+        """
+    } catch (Exception e) {
+        echo "skip dc/svc/route cleanup related exception, the resource may not exist. " + e.getMessage();
+    }
+    try {
+        sh """ 
+            oc create dc ${svcName} --image=${registry}/${imageNameSpace}/${svcName}:${tagName} -n ${projName} 2> /dev/null     
+            oc env dc ${svcName} APP_NAME=${svcName} -n ${projName} 2> /dev/null 
+            oc deploy ${svcName} --cancel -n ${projName} 2> /dev/null 
+            oc expose dc ${svcName} --port=8080 -n ${projName} 2> /dev/null 
+            oc expose svc ${svcName} --name=${svcName} -n ${projName} 2> /dev/null 
+        """
     } catch (Exception e) {
       echo "skip dc/svc/route creation related exception, the resource may already exist. " + e.getMessage();
     }
 
 }
 def promoteService (imageNamespace, projName, svcName, sourceTag, destinationTag) {
-    openshiftTag(namespace: imageNamespace,
-                  srcStream: svcName,
-                  srcTag: sourceTag,
-                  destStream: svcName,
-                  destTag: destinationTag)
-
-    openshiftDeploy(namespace: projName,
-  			     deploymentConfig: svcName,
-			     waitTime: '300000')
-
+    script {
```

Comment:
So everything required to put inside script {} tag this is only for using open shift plugin right ? is anything we required to change 3scale pipeline scipts ?

Reply:
script tag is need to run Groovy like statement.

Reply:
so it mean change required for 3scale scripts as well for DSL version. ? let me know if you will do that or I will make these changes in next PR.

Reply:
don't get your question.
but if your pipeline runs with no error, no script tag is needed.

Reply:
yes it runs without adding any scripts tag.  Hence resolving the conversation.